### PR TITLE
docs: architecture docs overhaul

### LINKED ISSUE #105: Architecture doc update
Things have evolved quite a bit since `architecture.md` was written, please do a pass to update it

### PR DESCRIPTION
**Pull request by andy-haynes**
This PR updates the architecture documentation with current explanations and diagrams.

Still need to finalize the callbacks and trust sections but please weigh in with thoughts ❤️ 

Fixes #105

### CODE REVIEW

File: architecture.md [hunk 1]
```diff
@@ -1,205 +1,42 @@
...
-   1. Serialize DOM subtree & props
-   2. Build set of child Components to fetch, creating a unique identifier for each
-   3. Post message to outer window
-      1. Render serialized DOM subtree (including DOM placeholders for child Components) in outer window
-      2. Initialize iframes for child Components (each one forking new workflow starting from #1)
+A root Component is only re-rendered in the outer application when the container requests it explicitly. Render requests
+are made when:
+- the container has finished initializing
+- state has changed within the Component (e.g. from a `useState` setter)
+- the root Component's `props` have changed, i.e. its parent Component re-rendered
```
Comment (mpeterdev) [inline architecture.md:27]: can this be simplified to say that it's based on when the Preact engine inside the iframe runs the render function again and detects dom changes, then give these two points as examples of when that frequently happens?

**Review by mpeterdev — COMMENTED** [review]
LGTM so far. Nice explanations, and the visuals are really helpful

**Comment by andy-haynes** [comment]
I still need to finish the `Callbacks` section but I wanted to push up what I had. Follow up PR in progress 🙂

**Comment by charleslavon** [comment]
Very helpful. Thanks for updating these docs @andy-haynes

**Pull request merged**